data/lua/stun.lua

```diff
@@ -0,0 +1,72 @@
+
+local _ = wesnoth.textdomain "wesnoth-units"
+local on_event = wesnoth.require("on_event")
+local old_unit_status = wesnoth.interface.game_display.unit_status
+
+function wesnoth.interface.game_display.unit_status()
+    local u = wesnoth.interface.get_displayed_unit()
+    if not u then return {} end
+    local s = old_unit_status()
+
+    if u.status.stunned then
+        table.insert(s, wml.tag.element{
+            image = "misc/stunned-status-icon.png",
+            tooltip = _"stunned: This unit is stunned. It cannot enforce its Zone of Control."
+        })
+    end
+
+    return s
+end
+
+local function on_hit(weapon, opponent)
+    if not weapon:matches{special_id = 'stun'} then return end
+    if not opponent.zoc then return end
+    local text
+    if opponent.gender == "female" then
+        text = _ "stunned"
```

Comment:
Genders are swapped compared to the `if` condition.

Reply:
Oh wow, whoops.